src/analyser.cpp

```diff
@@ -0,0 +1,1509 @@
+    };
+
+    size_t mOrder = MAX_SIZE_T;
+    Type mType = Type::UNKNOWN;
+
+    std::vector<AnalyserInternalEquationPtr> mDependencies;
+
+    AnalyserEquationAstPtr mAst;
+
+    std::vector<AnalyserInternalVariablePtr> mVariables;
+    std::vector<AnalyserInternalVariablePtr> mOdeVariables;
+
+    AnalyserInternalVariablePtr mVariable;
+    ComponentPtr mComponent = nullptr;
+
+    bool mComputedTrueConstant = true;
+    bool mComputedVariableBasedConstant = true;
+
+    bool mIsStateRateBased = false;
+
+    explicit AnalyserInternalEquation(const ComponentPtr &component);
+
+    void addVariable(const AnalyserInternalVariablePtr &variable);
+    void addOdeVariable(const AnalyserInternalVariablePtr &odeVariable);
+
+    static bool containsNonUnknownVariables(const std::vector<AnalyserInternalVariablePtr> &variables);
+    static bool containsNonConstantVariables(const std::vector<AnalyserInternalVariablePtr> &variables);
+
+    static bool knownVariable(const AnalyserInternalVariablePtr &variable);
+    static bool knownOdeVariable(const AnalyserInternalVariablePtr &odeVariable);
+
+    bool check(size_t & equationOrder, size_t & stateIndex, size_t & variableIndex);
+};
+
+AnalyserInternalEquation::AnalyserInternalEquation(const ComponentPtr &component)
+    : mAst(AnalyserEquationAst::create())
+    , mComponent(component)
+{
+}
+
+void AnalyserInternalEquation::addVariable(const AnalyserInternalVariablePtr &variable)
+{
+    if (std::find(mVariables.begin(), mVariables.end(), variable) == mVariables.end()) {
+        mVariables.push_back(variable);
+    }
+}
+
+void AnalyserInternalEquation::addOdeVariable(const AnalyserInternalVariablePtr &odeVariable)
+{
+    if (std::find(mOdeVariables.begin(), mOdeVariables.end(), odeVariable) == mOdeVariables.end()) {
+        mOdeVariables.push_back(odeVariable);
+    }
+}
+
+bool AnalyserInternalEquation::containsNonUnknownVariables(const std::vector<AnalyserInternalVariablePtr> &variables)
+{
+    return std::find_if(variables.begin(), variables.end(), [](const AnalyserInternalVariablePtr &variable) {
+               return (variable->mType != AnalyserInternalVariable::Type::UNKNOWN);
+           })
+           != std::end(variables);
+}
+
+bool AnalyserInternalEquation::containsNonConstantVariables(const std::vector<AnalyserInternalVariablePtr> &variables)
+{
+    return std::find_if(variables.begin(), variables.end(), [](const AnalyserInternalVariablePtr &variable) {
+               return (variable->mType != AnalyserInternalVariable::Type::UNKNOWN)
+                      && (variable->mType != AnalyserInternalVariable::Type::CONSTANT)
+                      && (variable->mType != AnalyserInternalVariable::Type::COMPUTED_TRUE_CONSTANT)
+                      && (variable->mType != AnalyserInternalVariable::Type::COMPUTED_VARIABLE_BASED_CONSTANT);
+           })
+           != std::end(variables);
+}
+
+bool AnalyserInternalEquation::knownVariable(const AnalyserInternalVariablePtr &variable)
```

Comment:
should be `isKnownVariable`?

Reply:
Agreed (and the same with `knownOdeVariable()` -> `isKnownOdeVariable()`). Fixed in commit 17707a82.